README.md
<!--
This README describes the package. If you publish this package to pub.dev,
this README's contents appear on the landing page for your package.

For information about how to write a good package README, see the guide for
[writing package pages](https://dart.dev/guides/libraries/writing-package-pages).

For general information about developing packages, see the Dart guide for
[creating packages](https://dart.dev/guides/libraries/create-library-packages)
and the Flutter guide for
[developing packages and plugins](https://flutter.dev/developing-packages).
-->

# edgy_button - Pin Edged Button in Flutter

A new and unique style of button that will take your Flutter Project to a next level

## Demo

![screenshot](https://user-images.githubusercontent.com/72114434/190898191-784904a5-667e-4467-975b-fb60cb2fe1a7.png)

## Getting started

   - In pubspec.yaml file, add the following dependency:

        ```
          edgy_button: ^0.0.1
        ```
   - Import the package into your project:
        ```
          import 'package:edgy_button/edgy_button.dart';
        ```
   - You can now use the `EdgyButton()` anywhere in the project
        ```
          // For normal Button
          EdgyButton(
                 buttonHeight: 50,
                 buttonWidth: 200,
                 label: 'BUTTON',
                 labelStyle: const TextStyle(
                   fontWeight: FontWeight.bold,
                   fontSize: 24,
                   color: Colors.white,
                 ),
                 buttonColor: Colors.redAccent,
                 onPressed: (){
                   log('Clicked');
                 },
             ),
             
           // For bordered Button
           EdgyButton(
               buttonHeight: 50,
               buttonWidth: 200,
               label: 'BUTTON',
               labelStyle: const TextStyle(
                 fontWeight: FontWeight.bold,
                 fontSize: 24,
                 color: Colors.green,
               ),
               border: Border.all(
                 color: Colors.green,
                 width: 2,
               ),
               buttonColor: Colors.transparent,
               onPressed: (){
                 log('Clicked Pro');
               },
             ),
        ```

## Features

| Property                | Description              |
|-------------------------|--------------------------|
| buttonHeight (required) | Height of the Button     |
| buttonWidth  (required) | Width of the Button      |
| label        (required) | Text inside the Button   |
| labelstyle              | For styling the text     |
| buttonColor  (required) | Color of the Button      |
| onPressed    (required) | onPressed Function       |
| border                  | Add border to the Button |


CHANGELOG.md
## [0.0.1] - 2022-09-18

Pin Edged Button in Flutter


test/edgy_button_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:edgy_button/edgy_button.dart';

void main() {
  var buttonHeight;
  var buttonWidth;
  var label;
  var buttonColor;
  var onPressed;
  EdgyButton(buttonHeight: buttonHeight, buttonWidth: buttonWidth, label: label, buttonColor: buttonColor, onPressed: onPressed);
}


lib/edgy_button.dart
library edgy_button;

import 'package:flutter/material.dart';

class EdgyButton extends StatefulWidget {

  final double buttonHeight;
  final double buttonWidth;
  final String label;
  final TextStyle? labelStyle;
  final Color buttonColor;
  final Border? border;
  final VoidCallback? onPressed;

  const EdgyButton({Key? key, required this.buttonHeight, required this.buttonWidth, required this.label, this.labelStyle, required this.buttonColor, required this.onPressed, this.border}) : super(key: key);

  @override
  State<EdgyButton> createState() => _EdgyButtonState();
}

class _EdgyButtonState extends State<EdgyButton> {
  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      onTap: widget.onPressed,
      child: Container(
        height: widget.buttonHeight,
        width: widget.buttonWidth,
        decoration: BoxDecoration(
          color: widget.buttonColor,
          border: widget.border,
          borderRadius: const BorderRadius.only(
            topRight: Radius.circular(15),
            bottomLeft: Radius.circular(15),
          ),
        ),
        child: Center(
          child: Text(
            widget.label,
            style: widget.labelStyle,
          ),
        ),
      ),
    );
  }
}


example/lib/main.dart
import 'dart:developer';

import 'package:flutter/material.dart';
import 'package:edgy_button/edgy_button.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'EdgyButton Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key});

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {

  @override
  Widget build(BuildContext context) {
    return Scaffold(
       body: Center(
         child: Column(
           mainAxisAlignment: MainAxisAlignment.center,
           children: [
             EdgyButton(
                 buttonHeight: 50,
                 buttonWidth: 200,
                 label: 'BUTTON',
                 labelStyle: const TextStyle(
                   fontWeight: FontWeight.bold,
                   fontSize: 24,
                   color: Colors.white,
                 ),
                 buttonColor: Colors.redAccent,
                 onPressed: (){
                   log('Clicked');
                 },
             ),
             const SizedBox(height:20),
             EdgyButton(
               buttonHeight: 50,
               buttonWidth: 200,
               label: 'BUTTON',
               labelStyle: const TextStyle(
                 fontWeight: FontWeight.bold,
                 fontSize: 24,
                 color: Colors.green,
               ),
               border: Border.all(
                 color: Colors.green,
                 width: 2,
               ),
               buttonColor: Colors.transparent,
               onPressed: (){
                 log('Clicked Pro');
               },
             ),
           ],
         ),
       ),
    );
  }
}
